### PR TITLE
I made my homework

### PR DESCRIPTION
**Pull request by Kani103**

### CODE REVIEW

File: week3/homework/src/index.js [hunk 1]
```diff
@@ -1,3 +1,65 @@
...
+const app = express();
+let todos = require('./todos')
+
+
+
+app.get('/', (req, res) => res.json(todos));
+
+
+app.get('/todos/:id', (req, res) => {
+    let found = todos.find(todo => {
```
Comment (eidosam) [inline week3/homework/src/index.js]: This variable is not re-assigned, so you need to use `const` for declaration.

File: week3/homework/src/index.js [hunk 2]
```diff
@@ -1,3 +1,65 @@
...
+        return todo.id === parseInt(req.params.id);
+    });
+
+    if (found) {
+        res.status(200).json(found);
+    } else {
+        res.send({ msg: '404 Not found'});
```
Comment (eidosam) [inline week3/homework/src/index.js]: You need to send status code 404 also

File: week3/homework/src/index.js [hunk 3]
```diff
@@ -1,3 +1,65 @@
...
+    });
+  
+    if (found) {
+      let targetIndex = todo.indexOf(found);
+      todo.splice(targetIndex, 1);
+    }
+      res.json({ result: "item deleted!"})
+      res.send(200);
+
+  });
```
Comment (eidosam) [inline week3/homework/src/index.js]: I guess this route supposed to remove all todos, not a single one, so you need to re-write empty array to the data file `./todos.json`.

But if you meant to remove single one, the path would be `'/todos'/:id`, and you need to re-write `todos` object to the file after `todos.splice`

File: week3/homework/src/index.js [hunk 4]
```diff
@@ -1,3 +1,65 @@
...
+
+  app.post('/todos/:id/done', (req, res) => {
+     let todoItem = todos.find(item => item.id === Number(req.params.id))
```
Comment (eidosam) [inline week3/homework/src/index.js]: `const`

File: week3/homework/src/index.js [hunk 5]
```diff
@@ -1,3 +1,65 @@
...
+     todoItem.done = true;
```
Comment (eidosam) [inline week3/homework/src/index.js]: You need to check if the item is found before updating it

File: week3/homework/src/index.js [hunk 6]
```diff
@@ -1,3 +1,65 @@
...
+     todos = todos.map(item => {
+         if (todoItem.id === item.id) {
+            return todoItem
+         }
+         return item
+     })
```
Comment (eidosam) [inline week3/homework/src/index.js]: You need to re-write the updated todos to the data file `./todos.json`

File: week3/homework/src/index.js [hunk 7]
```diff
@@ -1,3 +1,65 @@
...
+app.delete('/todos/:id/done', (req, res) => {
+    let todoItem = todos.find(item => item.id === Number(req.params.id))
+     todoItem.done = false;
+     todos = todos.filter(item => {
+         if (todoItem.id === item.id) {
+            return todoItem
+         }
+         return item
+     })
+     res.status(200)
```
Comment (eidosam) [inline week3/homework/src/index.js]: Same here, write the updated todos to the file

**Comment by wouterkleijn** [comment]
Thanks @eidosam ! And thanks for submitting your hw @Kani103  , please have a look at the comments of Samir. Once you have addressed it you are done :)

**Comment by wouterkleijn** [comment]
@Kani103 Kind reminder

**Pull request closed without merging**